pkgs/development/compilers/llvm/11/compiler-rt/default.nix

```
@@ -25,19 +25,20 @@ stdenv.mkDerivation {
     "-DCOMPILER_RT_DEFAULT_TARGET_ONLY=ON"
     "-DCMAKE_C_COMPILER_TARGET=${stdenv.hostPlatform.config}"
     "-DCMAKE_ASM_COMPILER_TARGET=${stdenv.hostPlatform.config}"
-  ] ++ lib.optionals (useLLVM || isDarwin || bareMetal || isMusl) [
+  ] ++ lib.optionals (useLLVM || bareMetal || isMusl ||
+                      (stdenv.hostPlatform.isDarwin && stdenv.hostPlatform.isAarch64)) [
     "-DCOMPILER_RT_BUILD_SANITIZERS=OFF"
     "-DCOMPILER_RT_BUILD_XRAY=OFF"
     "-DCOMPILER_RT_BUILD_LIBFUZZER=OFF"
     "-DCOMPILER_RT_BUILD_PROFILE=OFF"
-  ] ++ lib.optionals ((useLLVM || isDarwin || bareMetal) && !haveLibc) [
+  ] ++ lib.optionals ((useLLVM || bareMetal) && !haveLibc) [
     "-DCMAKE_C_COMPILER_WORKS=ON"
     "-DCMAKE_CXX_COMPILER_WORKS=ON"
     "-DCOMPILER_RT_BAREMETAL_BUILD=ON"
     "-DCMAKE_SIZEOF_VOID_P=${toString (stdenv.hostPlatform.parsed.cpu.bits / 8)}"
-  ] ++ lib.optionals ((useLLVM || isDarwin) && !haveLibc) [
+  ] ++ lib.optionals (useLLVM && !haveLibc) [
```

```suggestion
  ] ++ lib.optionals (!haveLibc) [
```
@s1341 I think this will still be fine for android? No reason not to link bionic right?

---

I think so.

---

`aarch64-android` is using llvm 12 anyway, but I imagine you'll port this change to 12 too.